src/stores/social-login-store.ts

```diff
@@ -19,58 +19,66 @@ export class SocialLoginStore {
     this.rootStore = rootStore;
 
     GoogleSignin.configure({
-      //   scopes: ['https://www.googleapis.com/auth/drive.readonly'], // what API you want to access on behalf of the user, default is email and profile
-      //   webClientId: '<FROM DEVELOPER CONSOLE>', // client ID of type WEB for your server (needed to verify user ID and offline access)
-      //   offlineAccess: false, // if you want to access Google API on behalf of the user FROM YOUR SERVER
-      //   hostedDomain: '', // specifies a hosted domain restriction
-      //   loginHint: '', // [iOS] The user's ID, or email address, to be prefilled in the authentication UI if possible. [See docs here](https://developers.google.com/identity/sign-in/ios/api/interface_g_i_d_sign_in.html#a0a68c7504c31ab0b728432565f6e33fd)
-      forceCodeForRefreshToken: true, // [Android] related to `serverAuthCode`, read the docs link below *.
-      //   accountName: '', // [Android] specifies an account name on the device that should be used
-      iosClientId:
-        '153461499435-0kruiebmi9ikjpg4fv531eli76oe1kgr.apps.googleusercontent.com', // [iOS] optional, if you want to specify the client ID of type iOS (otherwise, it is taken from GoogleService-Info.plist)
+      forceCodeForRefreshToken: true,
+      // iosClientId:
+      //   '153461499435-0kruiebmi9ikjpg4fv531eli76oe1kgr.apps.googleusercontent.com',
     });
   }
 
-  public async googleLogin() {
+  public async googleLogin(): Promise<LoginResult> {
     try {
-      console.log('user info');
       //   await GoogleSignin.hasPlayServices();
       const userInfo = await GoogleSignin.signIn();
+      const firstName = userInfo.user.givenName;
+      const lastName = userInfo.user.familyName;
+      // const email = userInfo.email;
       console.log(userInfo);
-      return true;
+      const data = await UsersApi.loginGoogle(userInfo.idToken);
+      console.log(data);
+      this.rootStore.profileFlowStore.firstName = firstName;
+      this.rootStore.profileFlowStore.lastName = lastName;
+
+      return data;
     } catch (error) {
       if (error.code === statusCodes.SIGN_IN_CANCELLED) {
-        // user cancelled the login flow
+        return null;
       } else if (error.code === statusCodes.IN_PROGRESS) {
         // operation (e.g. sign in) is in progress already
       } else if (error.code === statusCodes.PLAY_SERVICES_NOT_AVAILABLE) {
         // play services not available or outdated
       } else {
         // some other error happened
```

Comment:
?

Reply:
I need to take care of those cases... later